examples/rectangle.rs

```diff
@@ -45,16 +45,24 @@ fn main() {
         match event {
             Event::WindowEvent {
                 window_id,
-                event: WindowEvent::RedrawRequested,
+                event: WindowEvent::Resized(size),
             } if window_id == window.id() => {
-                // Grab the window's client area dimensions
-                if let (Some(width), Some(height)) = {
-                    let size = window.inner_size();
-                    (NonZeroU32::new(size.width), NonZeroU32::new(size.height))
-                } {
-                    // Resize surface if needed
+                if let (Some(width), Some(height)) =
+                    { (NonZeroU32::new(size.width), NonZeroU32::new(size.height)) }
+                {
+                    // Resize surface
                     surface.resize(width, height).unwrap();
-
+                }
+            }
+            Event::WindowEvent {
+                window_id,
+                event: WindowEvent::RedrawRequested,
+            } if window_id == window.id() => {
+                // Grab the window's client area dimensions, and ensure they're valid
+                let size = window.inner_size();
```

Comment:
Seems a bit funky that we don't have a size query on `Surface`, is that intentional?

Reply:
Have opened https://github.com/rust-windowing/softbuffer/issues/237 for that.